pyiron_contrib/RDM/storagejob.py

```diff
@@ -0,0 +1,324 @@
+
+    @storage_type.setter
+    def storage_type(self, storage_type):
+        if storage_type not in self._available_storage_types:
+            raise ValueError(f"Expected 'storage_type' to be one of {self._available_storage_types} "
+                             f"but got {storage_type}.")
+        self._storage_type = storage_type
+        for storage in self._available_storage_types:
+            self[storage] = False
+        self[storage_type] = True
+
+
+class StorageJob(GenericJob):
+    """Job to store files associated with meta data either locally, or at a remote data service like s3."""
+
+    def __init__(self, project, job_name):
+
+        super().__init__(project, job_name)
+        self.server.run_mode.interactive = True
+        self._stored_files = DataContainer(table_name='stored_files')
+        self._storage_type = StorageType(storage_type='local', table_name="storage_type")
+        self._external_storage = None
+
+    def _before_generic_remove_child(self):
+        if self._storage_type.s3:
+            _remove_s3_working_directory(self._external_storage, self.path)
+
+    def check_setup(self):
+        """
+        this function is called in self._run_if_new() if self.server.run_mode.queue
+        i.e. self.server.run_mode = queue if one sets a queue
+        afterwards self.save() and self.run() is called again -> self.run_if_created  followed by (in this case always) 
+        self.run_if_scheduler 
+        run_if_scheduler is the function called to sent the job to the cluster, however, it already sends the 
+        stuff to the queue which is obviously not intended if this is used as ssh storage system.
+        TODO: 
+        - overwrite run_if_scheduler to not trigger a job to be run on the queing system. 
+        - make some file_handler class to allow for 'store/receive/delete file' with a uniform interface for
+            - local
+            - s3
+            - ssh
+        """
+        raise NotImplementedError("Storing files remotely is not yet supported.")
+
+    @property
+    def storage_type(self):
+        return self._storage_type.storage_type
+
+    def use_s3_storage(self, config=None, bucket_name=None, _only_warn=False):
+        external_storage = FileS3IO(config=config, path=self.path, bucket_name=bucket_name)
+        if len(external_storage.list_nodes() + external_storage.list_groups()) > 0:
+            if not _only_warn:
+                raise ValueError("Storage location not empty.")
+            else:
+                self._logger.warning("Storage NOT empty - Danger of data loss!")
+        self._external_storage = external_storage
+        self._storage_type.storage_type = 's3'
+        self._storage_type.create_group("s3_config")
+        self._storage_type.s3_config.config = config
+        self._storage_type.s3_config.bucket_info = self._external_storage.bucket_info
+        self._storage_type.s3_config.bucket_name = bucket_name
+
+    def _list_groups(self):
```

Comment:
Shouldn't this still return the `"storage_type"` table? I thought child data containers should appear as groups.

Reply:
Hmm... The `StorageJob` stores its `storage_type` in a `DataContainer` subclass. However, the job itself does not have a `group`, does it? I could of course provide the `self._storage_type` here. However, I do not see any user benefit of getting the config storage back? The only thing I would like here (and this could actually be implemented) would be a new 'folder' to store files here, i.e. another `StorageJob` where the path is extended by the name of the group.

Reply:
Agree, with @niklassiemer this would be the place have 'folders' of uploaded files, if necessary.